File: 771.Jewels and Stones.cc
// https://leetcode.com/problems/jewels-and-stones/

class Solution {
public:
    int numJewelsInStones(string J, string S) {
        int jewels = 0;
        for (char& j : J) 
        {
            jewels += count(S.begin(), S.end(), j);
        }
        return jewels;
    }
};

File: leetcode/601-900/751-800/771.Jewels and Stones.rs
// https://leetcode.com/problems/jewels-and-stones

use std::collections::HashSet;

impl Solution {
    pub fn num_jewels_in_stones(jewels: String, stones: String) -> i32 {
        let jwls = jewels.chars().collect::<HashSet<_>>();

        let mut matches = 0;
        for s in stones.chars() {
            if jwls.contains(&s) {
                matches += 1;
            }
        } 

        matches
    }
}